python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -3378,12 +3378,6 @@ def test_series_astype_null_cases():
         ),
     )
 
-    # string to other
-    assert_eq(
-        gd.Series([1, 2, None, 3], dtype="int32"),
-        gd.Series(["1", "2", None, "3"]).astype("int32"),
-    )
```

Comment:
This should still work, why is this removed?

Reply:
Right now with present set of changes(which bring pandas behavior) will error in this case. 
We are trying to type-cast a `None` to int32 which is not supported in pandas. For example:

```python
>>> s = pd.Series(["10",None])
>>> s
0      10
1    None
dtype: object
>>> s.astype('int')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/generic.py", line 5882, in astype
    dtype=dtype, copy=copy, errors=errors, **kwargs
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 581, in astype
    return self.apply("astype", dtype=dtype, **kwargs)
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/internals/managers.py", line 438, in apply
    applied = getattr(b, f)(**kwargs)
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/internals/blocks.py", line 559, in astype
    return self._astype(dtype, copy=copy, errors=errors, values=values, **kwargs)
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/internals/blocks.py", line 643, in _astype
    values = astype_nansafe(vals1d, dtype, copy=True, **kwargs)
  File "/conda/envs/cudf/lib/python3.7/site-packages/pandas/core/dtypes/cast.py", line 707, in astype_nansafe
    return lib.astype_intsafe(arr.ravel(), dtype).reshape(arr.shape)
  File "pandas/_libs/lib.pyx", line 547, in pandas._libs.lib.astype_intsafe
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

So do we want to support type-casting a `None`/`null` in string to int typecast?

Reply:
Yes but in Pandas that's because they don't have nullable integers (in their standard type). For us we should be able to handle string `null` --> int `null`.